src/benchmark_runner.cc

```diff
@@ -148,7 +193,11 @@ class BenchmarkRunner {
       run_results.file_report_aggregates_only =
           (b.aggregation_report_mode & internal::ARM_FileReportAggregatesOnly);
     }
+  }
+
+  Derived& getDerived() { return *static_cast<Derived*>(this); }
```

Comment:
ugh. really?

i think the issue i have is that i don't really understand the problem that you're trying to solve (still). which is completely my fault for not being able to focus on this with you.

I think you're trying to get to the point where every iteration is used when we have repetitions to avoid having statistics at the repetition level that apply to _averages_ across iterations. Is that correct?

It also seems like you're trying to do this in a way that is optional. Is that correct?

Reply:
> ugh. really?

The previous review note was about the comment "do not call me from constructor,
since i call virtual method". Now that problem is gone, since there are no more virtual calls.

> i think the issue i have is that i don't really understand the problem that you're trying to solve (still).

I agree.

> which is completely my fault for not being able to focus on this with you.

No, probably largely a my fault, failing to *explain* what this is doing.

> I think you're trying to get to the point where every iteration is used when we have repetitions to avoid having statistics at the repetition level that apply to averages across iterations. Is that correct?

Let me try to highlight the issue i'm trying to solve:
https://docs.google.com/spreadsheets/d/11aEs8L8c9SciHAfs30XMNyea2LEBCil8enFRAVVlvzg/edit?usp=sharing
I.e. i want to avoid this implicit averaging within the each repetition,
and keep all the measurements separate, and thus e.g. get the correct stddev,

Now, this is slightly more problematic to do than that, because we don't *have* those
`N` separate measurements, we just keep the timer running over all the `N` iterations.

> It also seems like you're trying to do this in a way that is optional. Is that correct?

Correct. I really really really don't want to pessimize the existing `BenchmarkRunner`.
There are *very* good reasons for it's current behavior.
The new alternative variant (#710) by-design has more overhead.

Reply:
I don't like either of the options here. A comment warning about virtuals from constructors, and 'roll-your-own' virtual by downcasting this are both problematic.

I feel like we should be able to get somewhere simpler.

Abstracting BenchmarkRunner to do the iteration management and running the benchmark SGTM. Now when it comes to storing and reporting, either a simple conditional on iteration mode or (if you must) a virtual method with concrete implementations for the two modes should be doable without calling that method from the constructor. Even if that means abstracting out the construction from the running of the benchmark. What am I missing?